OpenRA.Mods.Common/Traits/Power/CanPowerDown.cs

```diff
@@ -20,9 +20,13 @@ public class CanPowerDownInfo : UpgradableTraitInfo, Requires<PowerInfo>
 		public readonly bool CancelWhenDisabled = false;
 
 		public readonly string IndicatorImage = "poweroff";
-		public readonly string IndicatorSequence = "offline";
+		[SequenceReference("IndicatorImage")] public readonly string IndicatorSequence = "offline";
 
-		public readonly string IndicatorPalette = "chrome";
+		[PaletteReference] public readonly string IndicatorPalette = "chrome";
+
+		public readonly string NotificationType = "Sounds";
```

Comment:
I mean hard-coding it and having lot's of boiling plate code spaghetti code like:

```csharp
if (Info.PowerupNotification != null && disabled)
	Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Notification", PowerdownNotification, self.Owner.Faction.InternalName);

if (Info.PowerupNotification != null && !disabled)
	Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Notification", PowerupNotification, self.Owner.Faction.InternalName);

if (Info.PowerupSpeech != null && disabled)
	Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech", PowerdownSpeech, self.Owner.Faction.InternalName);

if (Info.PowerupSpeech != null && !disabled)
	Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech", PowerupSpeech, self.Owner.Faction.InternalName);
```

is not improving anything.